quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigSourceProvider.java

```diff
@@ -35,9 +40,40 @@
 
 	@Override
 	public List<IConfigSource> getConfigSources(IJavaProject project) {
-		return Arrays.asList(new YamlConfigSource(APPLICATION_YAML_FILE, project),
-				new YamlConfigSource(APPLICATION_YML_FILE, project),
-				new PropertiesConfigSource(APPLICATION_PROPERTIES_FILE, project, 250));
+		List<IConfigSource> quarkusConfigSources = new ArrayList<>(3);
+		collectStaticConfigSources(quarkusConfigSources, project);
+		collectProfileConfigSources(quarkusConfigSources, project);
+		return quarkusConfigSources;
+	}
+
+	private static void collectStaticConfigSources(List<IConfigSource> quarkusConfigSources, IJavaProject project) {
+		quarkusConfigSources.add(new YamlConfigSource(APPLICATION_YAML_FILE, project));
+		quarkusConfigSources.add(new YamlConfigSource(APPLICATION_YML_FILE, project));
+		quarkusConfigSources.add(new PropertiesConfigSource(APPLICATION_PROPERTIES_FILE, project, 250));
+	}
+
+	private static void collectProfileConfigSources(List<IConfigSource> quarkusConfigSources, IJavaProject project) {
+		IPath projectPath = project.getPath().makeAbsolute();
+		IPath resourcesPath = projectPath.append("target/classes");
```

Comment:
It will not work with gradle project which have a bin folder different from target folder. The main idea is:

 * loop for sources folder (ex: src/main/resources) to get application-$profile.properties (and perhaps yaml?). To get sources folder, I suggest you see https://github.com/eclipse/lsp4mp/blob/f9afc25b0ba3083251fd60cbca976bb3603d0750/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/AbstractConfigSource.java#L78
 * create ApplicationProfileConfigSource which will be search pplication-$profile.properties from output folder (bin,target, etc) like you did.

This collect is done the first time, but you need to track add/delete of application-$profile.properties to add/remove ApplicationProfileConfigSource from the list.

Reply:
I have implemented this. I need to manually test it when I have multiple projects open still.